src/Three.js

```diff
@@ -119,6 +119,91 @@ if ( Function.prototype.name === undefined && Object.defineProperty !== undefine
 
 }
 
+if ( Number.EPSILON === undefined ) {
+
+	Object.defineProperty( Number, 'EPSILON', { value: Math.pow( 2, -52 ) } );
+
+}
+
+if ( Number.isNaN === undefined ) {
+
+	// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN
+
+	Number.isNaN = function(value) {
+
+		return typeof value === 'number' && isNaN(value);
+
+	}
+
+}
+
+if ( Object.assign === undefined ) {
```

Comment:
I guess the part below is a bit of copy paste, but there is a lot of formatting errors according to the style guidelines